File: Number of Laser Beams in a Bank/Number of Laser Beams in a Bank.java
// Solution for Number of Laser Beams in a Bank in JAVA

class Solution {
    public int numberOfBeams(String[] bank) {
        int beams = 0, prev = 0;
        for (String row : bank) {
            int curr = 0;
            for (char c : row.toCharArray()) if (c == '1') curr++;
            if (curr > 0) {
                beams += prev * curr;
                prev = curr;
            }
        }
        return beams;
    }
}

File: Number of Laser Beams in a Bank/Number of Laser Beams in a Bank.py
# Solution for Number of Laser Beams in a Bank in PY

class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        res, prev = 0, 0
        for row in bank:
            curr = row.count('1')
            if curr:
                res += prev * curr
                prev = curr
        return res
